C-Plus-Plus/ds/stackParenthesisChecker.cpp

```diff
@@ -0,0 +1,61 @@
+#include <iostream>
+#include<stack>
+using namespace std;
+
+// returns true if opening parenthesis
+bool isOpening(char c){
+    return c=='('||c=='['||c=='{';
+}
+
+// returns false if closing parenthesis
+bool isClosing(char c){
+    return c==')'||c==']'||c=='}';
+}
+
+//returs the corresponding opening parenthesis of a closing parenthesis
+char makeOpening(char c){
+    if(c==')')
+        return '(';
+    else if(c==']')
+        return '[';
+    else if(c=='}')
+        return '{';
+}
+
+// function to check if input is balanced
+bool checkBalanced(char *exp) {
+    stack<char>s;
+    int i=0;
+    while(exp[i]!='\0'){
+        if(isOpening(exp[i])){
+            s.push(exp[i]);
+        }
+        if(isClosing(exp[i])){
+            if(s.empty())           //if not an opening bracket in  the beginning
```

Comment:
Use comments above the code. Not beside it. Also prefer to use Multi-Line Comments